src/Azure.Functions.Cli/Kubernetes/FuncKeys/FuncAppKeysHelper.cs

```diff
@@ -0,0 +1,141 @@
+            {
+                funcKeysKubernetesEnvironVariables.Add(AzureWebJobsKubernetesSecretNameEnvVariableName, $"secrets/{keysSecretCollectionName}");
+            }
+
+            return funcKeysKubernetesEnvironVariables;
+        }
+
+        public static void CreateFuncAppKeysVolumeMountDeploymentResource(IEnumerable<DeploymentV1Apps> deployments, string funcAppKeysSecretsCollectionName)
+        {
+            if (deployments?.Any() == false)
+            {
+                return;
+            }
+
+            var volume = new VolumeV1
+            {
+                Name = FuncAppKeysVolumeName
+            };
+
+            if (!string.IsNullOrWhiteSpace(funcAppKeysSecretsCollectionName))
+            {
+                volume.VolumeSecret = new VolumeSecretV1 { SecretName = funcAppKeysSecretsCollectionName };
+            }
+
+            //Mount the app keys as volume mount to the container at the path "/run/secrets/functions-keys"
+            foreach (var deployment in deployments)
+            {
+                deployment.Spec.Template.Spec.Volumes = new VolumeV1[] { volume };
+                deployment.Spec.Template.Spec.Containers.First().VolumeMounts = new ContainerVolumeMountV1[]
+                {
+                        new ContainerVolumeMountV1
+                        {
+                            Name = FuncAppKeysVolumeName,
+                            MountPath = KubernetesSecretsMountPath
+                        }
+                };
+            }
+        }
+
+        public static void FunKeysMessage(SecretsV1 existingKeysSecret, SecretsV1 newKeysSecret)
+        {
+            if (existingKeysSecret?.Data?.Any() == true || newKeysSecret?.Data?.Any() == true)
+            {
+                ColoredConsole.WriteLine("Http Functions:");
+            }
+
+            if (existingKeysSecret?.Data?.Any() == true)
+            {
+                var existingFunctionKeys = existingKeysSecret.Data.Where(item => item.Key.StartsWith("functions"));
+                PrintKeyOutputMessage(existingFunctionKeys, " # this didn't change");
+            }
+
+            if (newKeysSecret?.Data?.Any() == true)
+            {
+                var newFunctionKeys = newKeysSecret.Data.Where(item => item.Key.StartsWith("functions"));
+                PrintKeyOutputMessage(newFunctionKeys, " # this was added");
+            }
+        }
+
+        private static void PrintKeyOutputMessage(IEnumerable<KeyValuePair<string, string>> functionKeys, string keyMsg)
+        {
+            var keyOutputMsgTemplate = "http://[ip]/api/{0}?code={1}";
+            foreach (var funcKey in functionKeys)
+            {
+                var functionName = funcKey.Key.Split('.')[1];
+                var functionKey = Encoding.UTF8.GetString(Convert.FromBase64String(funcKey.Value));
+                ColoredConsole.WriteLine(string.Concat("\t", string.Format(keyOutputMsgTemplate, functionName, functionKey), keyMsg));
+            }
```

Comment:
This is also not really correct. You can't build the URL just like that because we allow the users to customize the route per function and per host. You must query the runtime to get the actual list of routes. See the [`AzureHelper.PrintFunctionsInfo()`](https://github.com/Azure/azure-functions-core-tools/blob/dev/src/Azure.Functions.Cli/Helpers/AzureHelper.cs#L350) for more details 

You should be able to resolve the actual IP by polling the service created in the command above. If the service was of type `LoadBalancer`, you should be able to just hit it directly using the master key. We can leave `ClusterIP` for now because you'll need to `port-forward` to it.